### PR TITLE
Adds new functionality to the mortar, fixes a bad runtime that deleted turfs with the rangefinder

### PR DESCRIPTION
**Pull request by MacBlaze1**
## About The Pull Request

This PR allows binoculars with range-finding capability to store coordinates and click on the mortar to input them automatically and also fixes a bug that made mortar explosions not work at the very center, TiviPlus helped greatly with this
Closes #4364 
## Why It's Good For The Game

Bugs bad, and also this functionality works towards Rens goal of stopping FOB camping

## Changelog
:cl:
feature: Mortar crewmen with rangefinders can now instantly input coords in via clicking on the mortar with a rangefinder that has been used to target something.
fix: fixed range-finding causing turfs to be deleted.
/:cl:

### CODE REVIEW

File: code/game/objects/items/binoculars.dm [hunk 1]
```diff
@@ -26,13 +26,12 @@
...
 	var/obj/effect/overlay/temp/laser_target/laser
-	var/obj/effect/overlay/temp/laser_coordinate/coord
 	var/target_acquisition_delay = 100 //10 seconds
 	var/mode = 0 //Able to be switched between modes, 0 for cas laser, 1 for finding coordinates, 2 for directing railgun, 3 for orbital bombardment.
 	var/changable = TRUE //If set to FALSE, you can't toggle the mode between CAS and coordinate finding
 	var/ob_fired = FALSE // If the user has fired the OB
-	var/turf/current_turf // The target turf
-
+	var/turf/current_turf // The target turf, used for OBs
+	var/turf/target_turf // The target turf, used for mortars
```
Comment (TiviPlus) [inline code/game/objects/items/binoculars.dm:34]: Can you be a dear and convert the var comments to autodoc please

File: code/game/objects/structures/mortar.dm [hunk 2]
```diff
@@ -207,7 +207,27 @@
...
+		if(get_dist(loc, T) < 7)
+			to_chat(user, "<span class='warning'>You cannot aim at this coordinate, it is too close to your mortar.</span>")
+			return
+		if((binoc.target_turf.x > world.maxx || binoc.target_turf.x < 0) || (binoc.target_turf.y > world.maxx || binoc.target_turf.y < 0))
+			to_chat(user, "<span class='warning'>You cannot aim at this coordinate, it is outside of the area of operations.</span>")
+			return
+		targ_x = binoc.target_turf.x
+		targ_y = binoc.target_turf.y
+		to_chat(user, "<span class='notice'>You use the RFID chip in [I] to automatically send the target information to [src].</span>")
+		to_chat(user, "<span class='notice'>[I]'s RFID chip inputs the coordinates: Longitude [targ_x], Latitude [targ_y] into [src].</span>")
```
Comment (TiviPlus) [inline code/game/objects/structures/mortar.dm:230]: Use one tochat for this

File: code/game/objects/structures/mortar.dm [hunk 3]
```diff
@@ -207,7 +207,27 @@
...
 		new /obj/item/mortar_kit(loc)
 		qdel(src)
+	else if(istype(I,/obj/item/binoculars/tactical))
 
+		if(busy)
+			to_chat(user, "<span class='warning'>Someone else is currently using [src].</span>")
+			return
+		var/obj/item/binoculars/tactical/binoc = I
+		
+		if(binoc.target_turf.x == null || binoc.target_turf.y ==null)
```
Comment (TiviPlus) [inline code/game/objects/structures/mortar.dm:217]: Don't you also need to have a check for whether an area is underground?

Comment (MacBlaze1) [reply code/game/objects/structures/mortar.dm:217]: can this stuff be underground?

File: code/game/objects/items/binoculars.dm (hunk 1 above)
Comment (MacBlaze1) [reply code/game/objects/items/binoculars.dm]: done

File: code/game/objects/structures/mortar.dm (hunk 3 above)
Comment (TiviPlus) [reply code/game/objects/structures/mortar.dm:217]: I mean as in do you need to have a check here for if the targetted turf is underground or not

File: code/game/objects/items/binoculars.dm [hunk 4]
```diff
@@ -24,16 +24,22 @@
 	name = "tactical binoculars"
 	desc = "A pair of binoculars, with a laser targeting function. Ctrl+Click to target something."
 	var/laser_cooldown = 0
-	var/cooldown_duration = 200 //20 seconds
+	/// 20 seconds
```
Comment (spookydonut) [inline code/game/objects/items/binoculars.dm:27]: not a codedoc, you need to describe what the var is not the value of it

File: code/game/objects/items/binoculars.dm [hunk 5]
```diff
@@ -24,16 +24,22 @@
...
+	var/cooldown_duration = 200
```
Comment (spookydonut) [inline code/game/objects/items/binoculars.dm:28]:
```suggestion
	var/cooldown_duration = 20 SECONDS
```

File: code/game/objects/items/binoculars.dm [hunk 6]
```diff
@@ -24,16 +24,22 @@
...
 	var/obj/effect/overlay/temp/laser_target/laser
-	var/target_acquisition_delay = 100 //10 seconds
-	var/mode = 0 //Able to be switched between modes, 0 for cas laser, 1 for finding coordinates, 2 for directing railgun, 3 for orbital bombardment.
-	var/changable = TRUE //If set to FALSE, you can't toggle the mode between CAS and coordinate finding
-	var/ob_fired = FALSE // If the user has fired the OB
-	var/turf/current_turf // The target turf, used for OBs
-	///Last stored turf targetted by rangefinders
-	var/turf/targetturf
-
+	/// 10 seconds
```
Comment (spookydonut) [inline code/game/objects/items/binoculars.dm:31]: same

File: code/game/objects/items/binoculars.dm [hunk 7]
```diff
@@ -24,16 +24,22 @@
...
+	var/target_acquisition_delay = 100
```
Comment (spookydonut) [inline code/game/objects/items/binoculars.dm:32]:
```suggestion
	var/target_acquisition_delay = 10 SECONDS
```

File: code/game/objects/items/binoculars.dm [hunk 8]
```diff
@@ -24,16 +24,22 @@
...
-	var/mode = 0 //Able to be switched between modes, 0 for cas laser, 1 for finding coordinates, 2 for directing railgun, 3 for orbital bombardment.
-	var/changable = TRUE //If set to FALSE, you can't toggle the mode between CAS and coordinate finding
-	var/ob_fired = FALSE // If the user has fired the OB
-	var/turf/current_turf // The target turf, used for OBs
-	///Last stored turf targetted by rangefinders
-	var/turf/targetturf
-
+	/// 10 seconds
+	var/target_acquisition_delay = 100 
+	/// Able to be switched between modes, 0 for cas laser, 1 for finding coordinates, 2 for directing railgun, 3 for orbital bombardment.
```
Comment (spookydonut) [inline code/game/objects/items/binoculars.dm:33]: use a define for mode names

File: code/game/objects/items/binoculars.dm [hunk 9]
```diff
@@ -24,16 +24,22 @@
...
-	var/mode = 0 //Able to be switched between modes, 0 for cas laser, 1 for finding coordinates, 2 for directing railgun, 3 for orbital bombardment.
-	var/changable = TRUE //If set to FALSE, you can't toggle the mode between CAS and coordinate finding
-	var/ob_fired = FALSE // If the user has fired the OB
-	var/turf/current_turf // The target turf, used for OBs
-	///Last stored turf targetted by rangefinders
-	var/turf/targetturf
-
+	/// delay before your lazing action finishes and you successfully target something
+	var/target_acquisition_delay = 10 SECONDS
+	/// Able to be switched between modes, 0 for cas laser, 1 for finding coordinates, 2 for directing railgun, 3 for orbital bombardment.
```
Comment (spookydonut) [inline code/game/objects/items/binoculars.dm:33]:
```suggestion
	/// Able to be switched between modes, see [MODE_CAS]
```

**Comment by spookydonut** [comment]
tgui isnt built, do not tm

**Comment by TiviPlus** [comment]
closing as stale

**Pull request closed without merging**